sequence_processing_pipeline/ConvertJob.py

```diff
@@ -210,3 +222,152 @@ def parse_job_script(job_script_path):
+                             "has completed.")
+
+        project_names = list(self.info.keys())
+
+        # confirm source and destination projects are both valid.
+        for proj in [source_project, dest_project]:
+            if proj not in project_names:
+                raise ValueError(f"'{proj}' is not defined in sample sheet")
+
+        if source_project == dest_project:
+            raise ValueError(f"source '{source_project}' and destination "
+                             f"'{dest_project}' projects are the same")
+
+        # note that the user can supply a sample-name that didn't make it
+        # through the conversion step and may have no files matched to it.
+        # this is considered okay and a possible outcome of conversion. In
+        # this case zero files are copied and no information is passed back
+        # to the user.
+
+        # projects that contain replicates must also be considered. In this
+        # situation, the sample_name needs to be compared against all values
+        # in the orig_name column and the individual sample_names (w/well-ids)
+        # must be discovered; those individual sample_names can then be
+        # processed.
+
+        samples = self.info[source_project][SAMPLES_DETAILS_KEY]
+
+        # Either we will need to copy all replicates, in which case we need to
+        # match our input to orig_name values, or we want to copy
+        # a single sample, in which case the input needs to be matched to
+        # sample_name values. The results of our match will be
+        # stored in results and if we are not copying all replicates, there
+        # should only be one sample matched in results.
+        results = []
+
+        if self.contains_replicates is True:
+            for _, sample in samples.items():
+                # assume orig_name is present if copy_all_replicates is True.
+                if sample_name == sample[ORIG_NAME_KEY]:
+                    results.append(sample)
+        else:
+            # sample_name is a value from the sample_name column. it may or
+            # may not have a well-id appended and this sample-sheet may or
+            # may not contain replicates, but in either case a single sample
+            # either exists or it doesn't.
+            if sample_name in self.info[source_project][SAMPLES_DETAILS_KEY]:
+                results.append(samples[sample_name])
+
+        if len(results) == 0:
+            def _make_msg_base(col_name):
+                return (f"'{sample_name}' did not match any '{col_name}' "
+                        f"values in project '{source_project}'.")
+
+            if self.contains_replicates is True:
+                # the value of sample_name did not match any value in the
+                # orig_name column. It may match a value in the sample_name
+                # column.
+                msg = _make_msg_base(ORIG_NAME_KEY) +  \
+                      (f" Your value '{sample_name}' may have a well-id "
```

Comment:
This message seems oddly specific. Should we just test for that?

Reply:
If we have access to some ready-made function/regex that can identify well ids, I am up for that.  On the other hand, if we would have to generate that and add it in here, I prefer to leave it the way it is.  My thinking is that a human will be able to tell pretty much instantly from the sample name that is included in the error message whether a well id is present, and I am not comfortable trying to make the code recover from an erroneous sample name without a human intervention anyway.